### PR TITLE
Add a test that triggers a failure in Python

### PR DESCRIPTION
**Pull request by klecki**
## Description:
This is part of sanity test for CI configuration.
Using this test we can detect if CI correctly handles failing tests.

### Affected modules and functionalities:
CI/Tests

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [x] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/test/python/unittest_failure.cfg [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[unittest]
```
Comment (JanuszL) [inline dali/test/python/unittest_failure.cfg:1]: Do we need this config?

**Comment by klecki** [comment]
!build

File: qa/TL1_python-self-test-failure/test.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -e
+# used pip packages
+pip_packages='${python_test_runner_package}'
+target_dir=./dali/test/python
+
+test_body() {
+    ${python_new_invoke_test} --config unittest_failure.cfg test_trigger_failure.py
```
Comment (JanuszL) [inline qa/TL1_python-self-test-failure/test.sh:7]: How about just `python -c "raise TypeError()"`?

File: dali/test/python/unittest_failure.cfg (hunk 1 above)
Comment (klecki) [reply dali/test/python/unittest_failure.cfg:1]: So I can have a test case starting with `failure_` that won't be picked up by anyone that wants to run all the tests.
I also want to test the typical unit-testing scenario, that's why I want a test file and nose invocation.

File: qa/TL1_python-self-test-failure/test.sh (hunk 2 above)
Comment (klecki) [reply qa/TL1_python-self-test-failure/test.sh:7]: I can add another one, but I think it's better to try to test something that is as close as possible to the proper testing behaviour.

**Comment by klecki** [comment]
The other part is the MR 520 in CI.

File: qa/TL1_python-self-test-failure/test.sh (hunk 2 above)
Comment (JanuszL) [reply qa/TL1_python-self-test-failure/test.sh:7]: OK

Comment (mzient) [reply qa/TL1_python-self-test-failure/test.sh:7]: Mixed feelings here. I'd perhaps go as far the other way as possible and made the `test_body` execute something trivially failing, such as `rm /tmp/this_file_does_not_exist`.

**Pull request merged**